### PR TITLE
Avoid Cloning Buttons In SimpleFormIterator

### PR DESCRIPTION
**Pull request by djhi**

### CODE REVIEW

File: docs/Upgrade.md [hunk 1]
```diff
@@ -928,6 +928,35 @@ If you were using this prop, you can safely remove it.
...
+If you relied on those props in your custom buttons, you can should now leverage the following hooks:
+
+- `useSimpleFormIterator` for buttons that are not tied to an item such as the add button.
+
+```diff
+- import { Button, ButtonProps } from 'react-admin';
++ import { Button, ButtonProps, useSimpleFormIterator } from 'react-admin';
+
+export const MyAddButton = (props: ButtonProps) => {
++    const { add } = useSimpleFormIterator();
```
Comment (fzaninotto) [inline docs/Upgrade.md:943]: there is something missing in the diff: where did `add` come from in the past?

File: docs/Upgrade.md [hunk 2]
```diff
@@ -928,6 +928,35 @@ If you were using this prop, you can safely remove it.
...
+```
+
+- `useSimpleFormIteratorItem` for buttons that are tied to an item such as the remove and reorder buttons.
+
+```diff
+- import { Button, ButtonProps } from 'react-admin';
++ import { Button, ButtonProps, useSimpleFormIteratorItem } from 'react-admin';
+
+export const MyRemoveButton = (props: ButtonProps) => {
++    const { remove } = useSimpleFormIteratorItem();
```
Comment (fzaninotto) [inline docs/Upgrade.md:955]: same

**Review by fzaninotto — CHANGES_REQUESTED** [review]

File: docs/Upgrade.md [hunk 3]
```diff
@@ -928,6 +928,37 @@ If you were using this prop, you can safely remove it.
 
 The `data-generator-retail` package has been updated to provide types for all its records. In the process, we renamed the `commands` resource to `orders`. Accordingly, the `nb_commands` property of the `customers` resource has been renamed to `nb_orders` and the `command_id` property of the `invoices` and `reviews` resources has been renamed to `order_id`.
 
+## `<SimpleFormIterator>` No Longer Clones Its Buttons
+
+`<SimpleFormIterator>` used to clones the add, remove and reorder buttons and inject some props to them such as `onClick` and `className`.
+If you relied on those props in your custom buttons, you can should now leverage the following hooks:
```
Comment (WiXSL) [inline docs/Upgrade.md:934]:
```suggestion
If you relied on those props in your custom buttons, you should now leverage the following hooks:
```

**Review by WiXSL — CHANGES_REQUESTED** [review]

File: docs/Upgrade.md (hunk 2 above)
Comment (djhi) [reply docs/Upgrade.md:956]: Fixed

**Pull request merged**